graylog2-server/src/main/java/org/graylog/plugins/views/search/validation/TokenCollectingQueryParser.java

```diff
@@ -16,13 +16,26 @@
  */
 package org.graylog.plugins.views.search.validation;
 
+import org.apache.commons.collections4.CollectionUtils;
 import org.apache.lucene.analysis.Analyzer;
+import org.apache.lucene.index.Term;
+import org.apache.lucene.queryparser.classic.ParseException;
 import org.apache.lucene.queryparser.classic.QueryParser;
+import org.apache.lucene.search.Query;
 
+import java.util.Collection;
+import java.util.HashSet;
+import java.util.Iterator;
+import java.util.LinkedHashMap;
 import java.util.List;
+import java.util.Map;
+import java.util.Set;
 
 public class TokenCollectingQueryParser extends QueryParser {
```

Comment:
With all the changes and bigger responsibility, doesn't this class deserve a unit test?

Reply:
I think the best approach is to maintain and extend the LuceneQueryParser. This defines clear input and output of the operation. The TokenCollectingQueryParser is just an internal wrapper where I don't see any meaningful unit tests. But if you have any idea for a good test, I'll be happy to implement that.

Reply:
I see the technical possibility to unit test the class in such a way. I am rather questioning if such a test makes sense and would be useful. The newSmth methods are useful only in connection with the parser and actual query parsing itself. And from there, it's easy to observe the behaviour, without exactly inspecting internals. We can test those methods with a bit of mocking, but it seems like a test only to have a test.

The LuceneQueryParser is more of an integration test in a unit-test form. The LuceneQueryParser class logic is rather trivial and it mainly integrates the parser and visitor. But in the same time, it tests all those components together, with clean boundaries, inputs and outputs. 

If we manage to write a readable test for the TokenCollectingQueryParser.java, that would answer the question "what is this class purpose", then I am for it. I just worry that it will become a mess of mocks and map.get calls to verify something internal. 

I am lacking the imagination here - so please, if you have the time and will, could you add (in this branch) a test for that class, that you believe would be good? Thank you!

Reply:
Tomas, I completely understand your concerns. I've pushed a unit test and we can decide if it is ok or not.